internal/repositories/postgresRepository.go

```diff
@@ -0,0 +1,71 @@
+package repositories
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/brotigen23/go-url-shortener/internal/model"
+	_ "github.com/lib/pq"
+)
+
+type PostgresRepository struct {
+	db *sql.DB
+}
+
+func NewPostgresRepository(stringConnection string) *PostgresRepository {
+	ret := &PostgresRepository{}
+	db, err := sql.Open("postgres", stringConnection)
+	if err != nil {
+		panic(err)
+	}
+	_, err = db.Exec(`CREATE TABLE IF NOT EXISTS Aliases("URL" VARCHAR UNIQUE, "Alias" VARCHAR)`)
+	if err != nil {
+		fmt.Println("Error: ", err)
+	}
+	ret.db = db
+
+	return ret
+}
+
+func (repo *PostgresRepository) CloseConnection() {
+	repo.db.Close()
+}
+
+func (repo *PostgresRepository) GetByAlias(alias string) (*model.Alias, error) {
+	query := repo.db.QueryRow(`SELECT * FROM Aliases WHERE "Alias" = $1`, alias)
+	var URL string
+	var Alias string
+	err := query.Scan(&URL, &Alias)
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+	return &model.Alias{URL: URL, Alias: Alias}, nil
+}
+func (repo *PostgresRepository) GetByURL(url string) (*model.Alias, error) {
+	query := repo.db.QueryRow(`SELECT * FROM Aliases WHERE "URL" = $1`, url)
+	var URL string
+	var Alias string
+	err := query.Scan(&URL, &Alias)
+	if err != nil {
+		fmt.Println(err)
+		return nil, err
+	}
+	return &model.Alias{URL: URL, Alias: Alias}, nil
+
+}
+
+func (repo *PostgresRepository) GetAll() *[]model.Alias { return nil }
+func (repo *PostgresRepository) Save(model model.Alias) error {
+	result, err := repo.db.Exec("INSERT INTO Aliases VALUES($1, $2)", model.URL, model.Alias)
```

Comment:
[CHANGE REQUEST] Please make a plan of this query.
https://www.postgresql.org/docs/current/using-explain.html